### PR TITLE
feat: add payment_sent notification

### LINKED ISSUE #426: Add payment_sent support for NIP-47 notifications
Currently NWC only supports the `payment_received` notification, but to be consistent and enable other possibilities, clients should also be able to be notified when LDK makes a payment.

- in NWC
- JS SDK
- copy changes for https://github.com/nostr-protocol/nips/pull/1164

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #426 | Adds support for `payment_sent` notification

- Have to add this to the NIP-47 docs, waiting for confirmation on the notification payload
- `js-sdk` PR: https://github.com/getAlby/js-sdk/pull/223

Testing
- [x] Added test in NWC
- [x] split tests - @im-adithya 
- [x] Check from rails app - @im-adithya 
- [ ] Update NIP-47 notifications PR - @rolznz - TODO afterwards.
- [x] Wait for response from Kukks to see we don't break something - @rolznz - yes, it will just fall back to polling (but I don't think that's the case anyway because it probably just uses the published info event)

### CODE REVIEW

**Comment by im-adithya** [comment]
To be added in https://github.com/nostr-protocol/nips/pull/1164
```
### `payment_sent`

Description: A payment was successfully sent from the wallet.

Notification:
```jsonc
{
    "notification_type": "payment_sent",
    "notification": {
        "type": "outgoing",
        "invoice": "string", // encoded invoice
        "description": "string", // invoice's description, optional
        "description_hash": "string", // invoice's description hash, optional
        "preimage": "string", // payment's preimage
        "payment_hash": "string", // Payment hash for the payment
        "amount": 123, // value in msats
        "fees_paid": 123, // value in msats
        "created_at": unixtimestamp, // invoice/payment creation time
        "expires_at": unixtimestamp, // invoice expiration time, optional if not applicable
        "settled_at": unixtimestamp, // invoice/payment settlement time
        "metadata": {} // generic metadata that can be used to add things like zap/boostagram details for a payer name/comment/etc.
    }
}
```
```

**Comment by rolznz** [comment]
@im-adithya can you update this PR?

File: nip47/notifications/nip47_notifier_test.go [hunk 1]
```diff
@@ -84,6 +84,26 @@ func TestSendNotification(t *testing.T) {
 	assert.Equal(t, tests.MockTransaction.Amount, transaction.Amount)
 	assert.Equal(t, tests.MockTransaction.FeesPaid, transaction.FeesPaid)
 	assert.Equal(t, tests.MockTransaction.SettledAt, transaction.SettledAt)
+
```
Comment (rolznz) [inline nip47/notifications/nip47_notifier_test.go:87]: Can you please try to add new tests - these become difficult to maintain. I had to split all the old ones which took a lot of time

File: nip47/notifications/nip47_notifier.go [hunk 2]
```diff
@@ -44,28 +44,54 @@ func NewNip47Notifier(relay Relay, db *gorm.DB, cfg config.Config, keys keys.Key
...
+	if event.Event != "nwc_payment_received" && event.Event != "nwc_payment_sent" {
 		return nil
 	}
 
-	paymentReceivedEventProperties, ok := event.Properties.(*events.PaymentReceivedEventProperties)
-	if !ok {
-		logger.Logger.WithField("event", event).Error("Failed to cast event")
-		return errors.New("failed to cast event")
-	}
+	var transaction interface{}
```
Comment (rolznz) [inline nip47/notifications/nip47_notifier.go:51]: I will refactor this a bit, it's missing a few things and does not consider notifications that are not transactions

File: events/models.go [hunk 3]
```diff
@@ -24,6 +24,13 @@ type PaymentReceivedEventProperties struct {
 	NodeType    string `json:"node_type"`
 }
 
+type PaymentSentEventProperties struct {
+	PaymentId   string `json:"payment_id"`
```
Comment (rolznz) [inline events/models.go:28]: I will remove everything except `payment_hash`. I don't think the others are needed (also trying to remove unnecessary properties from the PaymentReceivedEventProperties too)

**Comment by rolznz** [comment]
@im-adithya @bumi I looked at the rails code and it seems to look ok to handle both incoming and outgoing invoices without any changes

I'm merging this now. In case you have any feedback, let me know and I'll address it in a follow up PR

**Pull request merged**